src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseMessage.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Formats.Cbor;
+using System.Runtime.Versioning;
+
+namespace System.Security.Cryptography.Cose
+{
+    public abstract class CoseMessage
+    {
+        private const byte EmptyStringByte = 0xa0;
+        // COSE tags https://datatracker.ietf.org/doc/html/rfc8152#page-8 Table 1.
+        internal const CborTag Sign1Tag = (CborTag)18;
+
+        internal byte[]? _content;
+        internal byte[] _signature;
+        internal byte[] _protectedHeaderAsBstr;
+
+        private CoseHeaderMap _protectedHeaders;
+        private CoseHeaderMap _unprotectedHeaders;
+        public CoseHeaderMap ProtectedHeader => _protectedHeaders;
+        public CoseHeaderMap UnprotectedHeader => _unprotectedHeaders;
```

Comment:
These feel like they should be plural.

```suggestion
        public CoseHeaderMap ProtectedHeaders => _protectedHeaders;
        public CoseHeaderMap UnprotectedHeaders => _unprotectedHeaders;
```

Reply:
Yes, it should be plural or have a name that conveys "a set of headers". The [RFC](https://datatracker.ietf.org/doc/html/rfc8152#section-3) also names it "buckets".
So it can be:
* ProtectedHeaders
* ProtectedHeaderMap
* ProtectedBucket

Reply:
ProtectedHeaders should be good for now.